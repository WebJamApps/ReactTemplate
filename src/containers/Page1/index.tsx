import commonUtils from 'src/lib/commonUtils';

export const Page1 = (): JSX.Element => {
  commonUtils.setTitleAndScroll('Page 1', window.screen.width);
  return (
    <div>
      <h1>Page 1</h1>
      <div>
        <p>Some page one content here.</p>
      </div>
    </div>
  );
};
