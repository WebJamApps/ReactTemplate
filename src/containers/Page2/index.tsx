import { gql, useQuery } from '@apollo/client';
import commonUtils from 'src/lib/commonUtils';

export const TEST_GQL = gql`
  query Gqldocs {
    gqldocs {
      title
      author
    }
  }
`;

export const Page2 = (): JSX.Element => {
  const { loading, error, data } = useQuery(TEST_GQL);
  commonUtils.setTitleAndScroll('Page 2', window.screen.width);
  if (loading) return <p>Loading...</p>;
  if (error) {
    return (
      <p>
        Error :
        {' '}
        {error.message}
      </p>
    );
  }
  return (
    <div>
      <h1>Page 2</h1>
      <div>
        <p>Some page two content here.</p>
        {data ? <p>{JSON.stringify(data)}</p> : null}
      </div>
    </div>
  );
};
