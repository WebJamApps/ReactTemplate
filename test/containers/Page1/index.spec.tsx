import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Page1 } from 'src/containers/Page1';

describe('Page1', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<BrowserRouter><Page1 /></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
