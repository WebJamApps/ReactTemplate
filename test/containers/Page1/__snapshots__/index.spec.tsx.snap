// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page1 renders correctly 1`] = `
<div>
  <h1>
    Page 1
  </h1>
  <div>
    <p>
      Some page one content here.
    </p>
  </div>
</div>
`;
